include/circt/Dialect/SystemC/SystemCStructure.td

```diff
@@ -207,3 +207,167 @@ def DestructorOp : SystemCOp<"cpp.destructor", [
   let hasVerifier = true;
   let skipDefaultBuilders = 1;
 }
+
+def FuncOp : SystemCOp<"cpp.func", [
```

Comment:
I completely agree with you. It would be great to have a way in TableGen to inherit from an operation (creating no sub-type relation or such, but just to reuse the code) to add a few attributes and only have to deal with verification, etc. of those. Or alternatively have utilities in TableGen to easily create specialized FuncOps with their return and call operations (i.e. tablegen automatically generates the verification code to check the operands of those three ops match in terms of size and types, etc.).